### PR TITLE
Issue 2096/error handling for notes upload

### LINKED ISSUE #2096: Journey Instance Notes Fail to be added if long
## Description

Organisers are struggling to add notes to journeys when they are too long, with no feedback given. This can bea

## Steps to reproduce
1. Go to a journey instance
2. Click on add note
3. Write a very long note
4. Click submit

### Expected Behaviour

Either the note is accepted and added to the journey instance, or a warning is shown if it is too long (or whatever is happening)

### Actual Behaviour

The note is not added to the journey, and the submit button remains greyed out until you refresh the page.

## Screenshots (if you have any)

[Screencast from 31-07-24 16:06:57.webm](https://github.com/user-attachments/assets/0f8fffc0-9bb0-4e09-857c-69aa3dc9a985)

### PR DESCRIPTION
**Pull request by Jrende**
## Description
This PR makes it so you get an error message when trying to upload a too long journey note, instead of just annihilating your hard-written text.

## Screenshots
Screen Shot 2024-09-28 at 17 44 45

## Changes
* When the POST request to save a note fails, an error message is show. We don't know exactly why this happens, but we provide a helpful note.

## Related issues
Resolves #2096

### CODE REVIEW

File: src/zui/ZUITimeline/l10n/messageIds.ts [hunk 1]
```diff
@@ -12,6 +12,9 @@ export default makeMessages('zui.timeline', {
     },
   },
   expand: m('show full timeline'),
+  fileUploadErrorMessage: m(
+    "Unable to add note. Possible reason is that it's too long."
```
Comment (richardolsson) [inline src/zui/ZUITimeline/l10n/messageIds.ts:16]: I would rephrase this:

```suggestion
    "Unable to add note. It might be too long."
```

File: src/pages/organize/[orgId]/journeys/[journeyId]/[instanceId]/index.tsx [hunk 2]
```diff
@@ -74,6 +74,7 @@ const JourneyDetailsPage: PageWithLayout = () => {
   const { orgId, instanceId } = useNumericRouteParams();
   const messages = useMessages(messageIds);
   const [isLoading, setIsLoading] = useState(false);
+  const [showPostRequestError, setShowPostRequestError] = useState(false);
```
Comment (richardolsson) [inline src/pages/organize/[orgId]/journeys/[journeyId]/[instanceId]/index.tsx:77]: Would it make sense to move this state into `ZUITimeline` instead?

File: src/zui/ZUITimeline/index.tsx [hunk 3]
```diff
@@ -46,7 +48,11 @@ const ZUITimeline: React.FunctionComponent<ZUITimelineProps> = ({
     <Fade appear in timeout={1000}>
       <Grid container direction="column" spacing={5}>
         <Grid item>
-          <TimelineAddNote disabled={disabled} onSubmit={onAddNote} />
+          <TimelineAddNote
+            disabled={disabled}
+            onSubmit={onAddNote}
```
Comment (richardolsson) [inline src/zui/ZUITimeline/index.tsx:53]: If the state is moved in here, then this line could instead be:

```suggestion
            onSubmit={onAddNote}
            onSubmit={(note) => {
                try {
                    onAddNote(note);
                } catch (err) {
                    // Set state here
                }
            }
```

**Review by richardolsson — CHANGES_REQUESTED** [review]
Awesome work! I've tried to preview build to confirm that it works, and I've reviewed the code which looks mostly very good. I don't love how many components are involved though, so I'm wondering if it would make sense to keep the state within `ZUITimeline`. But I can totally be convinced otherwise!

I would like to change the text of the error message though. 👇

File: src/zui/ZUITimeline/l10n/messageIds.ts (hunk 1 above)
Comment (Jrende) [reply src/zui/ZUITimeline/l10n/messageIds.ts:16]: Fixed

File: src/zui/ZUITimeline/index.tsx (hunk 3 above)
Comment (Jrende) [reply src/zui/ZUITimeline/index.tsx]: Good suggestion, I've pushed a change where I moved the error handling

File: src/zui/ZUITimeline/l10n/messageIds.ts (hunk 1 above)
Comment (richardolsson) [reply src/zui/ZUITimeline/l10n/messageIds.ts:16]: Did you forget to commit this? I can't find it in the PR diff.

Comment (Jrende) [reply src/zui/ZUITimeline/l10n/messageIds.ts]: I did forget, but I've pushed now

**Review by richardolsson — APPROVED** [review]
I appreciate it! Great work on this. 👏
